File: Binary二进制/leetcode_868_binaryGap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        last = -1
        i = 0
        res = 0
        while n:
            if n&1:
                if last >= 0:
                    res = max(res, i-last)
                last = i
            n >>= 1
            i += 1
        return res

if __name__ == "__main__":
    s = Solution()
    n = 22
    print(s.binaryGap(n))

File: 02.Binary二进制/leetcode_868_binaryGap.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int binaryGap(int n) {
        int last = -1;
        int i = 0;
        int res = 0;
        while (n){
            if (n&1){
                if (last >= 0){
                    res = max(res, i-last);
                }
                last = i;
            }
            n >>= 1;
            i++;
        }
        return res;
    }
};

int main(){
    Solution s;
    int n = 22;
    cout << s.binaryGap(n) << endl;
}
